### ISSUE TITLE
Question on zfs-check

### ISSUE THREAD

**Issue by Simon4711** [psy0rz/zfs_autobackup#176]
### Discussed in https://github.com/psy0rz/zfs_autobackup/discussions/175

<div type='discussions-op-text'>

<sup>Originally posted by **Simon4711** November 30, 2022</sup>
I tried out the zfs-check tool, and I don't know how to see this failure:

root@pvezfsdus:/home# zfs-check rpool/crypt/hamdus1/USBBackup/Crypt/vm-100-disk-0@hamdus1-20221130030011 --skip 9 --debug | ssh root@pve3ham "zfs-check USBBackup/Crypt/vm-100-disk-0@hamdus1-20221130030011 --check"
**Chunk 56320 failed: da39a3ee5e6b4b0d3255bfef95601890afd80709 EOF**

I think this is the last generated hash on the volume. I tried access to this zfs vol generating a new one from this checked snapshot in Proxmox, it's a RAW in a Windows VM. It seems to be o.k.

But why is this error reported on zfs-check?</div>

**Comment by psy0rz** [comment]
Strange indeed!

An EOF would mean that USBBackup/Crypt/vm-100-disk-0@hamdus1-20221130030011 is too small.

But its probably a bug in zfs-check i hope, although there a quite a bit of regression tests to catch all the edge cases.

Could you make absolutely sure by comparing the sha1sum of both sides?

**Comment by psy0rz** [comment]
No, i mean check it without zfs-check to make sure that both volumes are exactly the same.

You should clone the snapshots on both sides, and then run sha1sum on the actual device:
```
zfs clone USBBackup/Crypt/vm-100-disk-0@hamdus1-20221130030011 USBBackup/test
sha1sum /dev/zvol/USBBackup/test

```

Do that on both sides and see if the checksum match. 

How big is that volume?

**Comment by Simon4711** [comment]
O.K. It's checksumming on both sides, lasts a bit.... I do that on another volume synced before, and same failure message of zfs-check. The volume is about 1 TB.

**Comment by psy0rz** [comment]
And what was the result of that one you did before?

**Comment by Simon4711** [comment]
This chunk failed error with EOF too. But lower chunk number, the one reported before is a 6 TB USB volume. The checksum I'm now calculating is on a normal SATA Z1 to have a quicker result.

**Comment by Simon4711** [comment]
Yes, SHA1 Checksum is the same for source and target vol:

root@pve3ham:~# sha1sum /dev/zvol/rpool/test
7b304236dc897b5423c6e5237a85a0b96507f4a9  /dev/zvol/rpool/test

root@pvezfsdus:~# sha1sum /dev/zvol/rpool/test
7b304236dc897b5423c6e5237a85a0b96507f4a9  /dev/zvol/rpool/test

**Comment by psy0rz** [comment]
Ok then its definitely a zfs-check bug.

**Comment by Simon4711** [comment]
Maybe cause I started zfs-check on the target? I checked 3 zfs vols with it from the same WinServer 2019 VM on proxmox, synced before with zfs-autobackup. The sizes and outputs are:

200 GB  > o.k. no message
1000 GB > EOF 
6000 GB > EOF
